File: lds-developv2/Frontend/angularapp/src/app/models/Rating.ts
export class Rating {

  constructor(
    public ratingID?: number, // Id da avaliação
    public userID?: number, // ID do utilizador que efetuou a avaliação
    public recintoID?: number, // Id do recinto que foi avaliado
    public username?: string, // nome do utilizador que efetuou a avaliação
    public rating?: number, // rate {1-5}
    public description?: string, // Descrição caso o utilizador queira
  ) { } 
}

File: lds-developv2/Backend/Models/Rating.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace Friendly_.Models
{
    public class Rating
    {
        [Key]
        public int ratingID { get; set; }

        public User? user { get; set; }
        [Required]
        public int userID { get; set; }

        public RecintoDesportivo? recinto { get; set; }
        [Required]
        public int recintoID { get; set; }

        [Required(ErrorMessage = "Especifique um Nome")]
        [MaxLength(100)]
        [DisplayName("Nome")]
        public String username { get; set; }

        public int rating { get; set; }


        [DisplayName("Descrição Recinto")]
        public String? description { get; set; }
    }
}
